controllers/masteruserrecord/masteruserrecord_controller.go

```diff
@@ -235,14 +304,8 @@ func (r *Reconciler) useExistingConditionOfType(condType toolchainv1alpha1.Condi
 }
 
 func (r *Reconciler) manageCleanUp(logger logr.Logger, mur *toolchainv1alpha1.MasterUserRecord) (time.Duration, error) {
-	for _, ua := range mur.Spec.UserAccounts {
-		requeueTime, err := r.deleteUserAccount(logger, ua.TargetCluster, mur.Name)
-		if err != nil {
-			return 0, r.wrapErrorWithStatusUpdate(logger, mur, r.setStatusFailed(toolchainv1alpha1.MasterUserRecordUnableToDeleteUserAccountsReason), err,
-				"failed to delete UserAccount in the member cluster '%s'", ua.TargetCluster)
-		} else if requeueTime > 0 {
-			return requeueTime, nil
-		}
+	if requeue, err := r.ensureUserAccountsAreNotPresent(logger, mur, r.membersWithoutUserAccount(nil)); err != nil || requeue > 0 {
```

Comment:
I'm not a fan of using reflect when not necessary needed.
Changed that as part of [aa08af8](https://github.com/codeready-toolchain/host-operator/pull/872/commits/aa08af89355ebe0263cd55e50c944d35b5063299) I hope that it will be better

Reply:
yeah I'm not a big fan of reflect either. I heard in one of the next Go versions we will have something like `map.Keys()` available as builtin. Thanks for changing that.